ingester/ingester.go

```diff
@@ -596,6 +602,14 @@ func (i *Ingester) flushUserSeries(userID string, fp model.Fingerprint, immediat
 	if len(series.chunkDescs) == 0 {
 		userState.fpToSeries.del(fp)
 		userState.index.delete(series.metric, fp)
+
+		metricName, err := util.ExtractMetricNameFromMetric(series.metric)
+		if err != nil {
+			// Series without a metric name should never be able to make it into
+			// the ingester's memory storage.
+			panic(err)
+		}
+		userState.decNumSeriesInMetric(metricName)
```

Comment:
Can you roll the above three things (`fpToSeries.del`, `index.delete` and `userState.decNumSeriesInMetric`) into a function on `userState` pls?